### PR TITLE
EDGAR 23.4 preview

### PR DESCRIPTION
**Pull request by hermfischer-wf**
#### Reason for change
EDGAR release 23.4 preview

#### Description of change
See https://www.sec.gov/edgar/filermanual

#### Steps to Test
See https://www.sec.gov/structureddata/osdinteractivedatatestsuite

To get GUI of EdgarRenderer to launch in ixviewer-plus mode or production 23.2 mode, see lines to uncomment in plugin/EdgarRenderer/resources/Summarize.xslt (uncomment line 19 or 22)

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/EFM/resources/us-gaap-calc-2020.json [hunk 1]
```diff
@@ -0,0 +1,14578 @@
+{
```
Comment (derekgengenbacher-wf) [inline arelle/plugin/validate/EFM/resources/us-gaap-calc-2020.json:1]: Why the changes in this file? GAAP 2020 isnt even supported by the SEC anymore?

**Comment by RamkumarAlagupandian** [comment]
@hefischer @austinmatherne-wk 

**Observation Regarding EDGAR 23.4 PR Behavior**

I would like to draw your attention to a notable behavior observed in the EDGAR 23.4 PR. While utilizing the specified command line to run and retrieve R files, I have encountered the generation of duplicate R files. What makes this situation intriguing is that, despite being duplicates, the content of these files appears to differ.

I am uncertain if this issue is widespread or specific to my instance. Recently, I have observed some code changes within the system itself, transitioning from locating ixds in zip files to identifying inline files. Given that the SEC is poised to deploy the EDGAR 23.4 version in their production system, I believe it is crucial to address this matter promptly.

To assist in understanding the issue, I have attached a sample file set that I utilized, along with screenshots depicting the duplicated rendering. 

**Command line -** 

arelleCmdLine.exe  --plugin EdgarRenderer --disclosureSystem efm-pragmatic -v -f "C:\\temp\\0022\\05_10K_Ex26"

image

[eelo00001gd-20340331_pre.zip](https://github.com/Arelle/Arelle/files/13526304/eelo00001gd-20340331_pre.zip)

Your insights and guidance on resolving this matter would be greatly appreciated.

Thank you for your attention to this concern, and I look forward to your valuable input.

**Comment by hermfischer-wf** [comment]
> @hefischer @austinmatherne-wk @hermfischer-wf
> 
> **Observation Regarding EDGAR 23.4 PR Behavior**
> 
> I would like to draw your attention to a notable behavior observed in the EDGAR 23.4 PR. While utilizing the specified command line to run and retrieve R files, I have encountered the generation of duplicate R files. What makes this situation intriguing is that, despite being duplicates, the content of these files appears to differ.
> 
> I am uncertain if this issue is widespread or specific to my instance. Recently, I have observed some code changes within the system itself, transitioning from locating ixds in zip files to identifying inline files. Given that the SEC is poised to deploy the EDGAR 23.4 version in their production system, I believe it is crucial to address this matter promptly.
> 
> To assist in understanding the issue, I have attached a sample file set that I utilized, along with screenshots depicting the duplicated rendering.
> 
> **Command line -**
> 
> arelleCmdLine.exe --plugin EdgarRenderer --disclosureSystem efm-pragmatic -v -f "C:\temp\0022\05_10K_Ex26"
> 

> [eelo00001gd-20340331_pre.zip](https://github.com/Arelle/Arelle/files/13526304/eelo00001gd-20340331_pre.zip)
> 
> Your insights and guidance on resolving this matter would be greatly appreciated.
> 
> Thank you for your attention to this concern, and I look forward to your valuable input.

I cannot reproduce the R5 output (where document type is shown for 3-month period).  However the attached zip has a file with EX-26/A document type.  There is no /A for EX-26 (it was erroneously allowed by an earlier version of the ex 26 rules).  I've changed to EX-26 locally (without /A) for testing and things seem to work right.

**Comment by hermfischer-wf** [comment]
EFM suite [efm-68d-231120.zip ](https://www.sec.gov/structureddata/osdinteractivedatatestsuite) with latest shr seems to have different results: [arelle:testSuiteResults] pass=5630, fail=0 - testcases.xml

**Review by austinmatherne-wk — DISMISSED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]
